File: PowMoji/app/src/main/java/pow/jie/powmoji/MainActivity.java
package pow.jie.powmoji;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;

import com.google.android.material.tabs.TabLayout;

import java.util.ArrayList;
import java.util.List;

import pow.jie.powmoji.adapter.MyPagerAdapter;
import pow.jie.powmoji.fragment.MojiFragment;

public class MainActivity extends AppCompatActivity {

    MyPagerAdapter adapter;
    private List<MojiFragment> fragments = new ArrayList<>();
    private List<String> titles = new ArrayList<>();
    TabLayout tabLayout;
    ViewPager viewPager;
    Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        toolbar = findViewById(R.id.tb_main);
        setSupportActionBar(toolbar);

        viewPager = findViewById(R.id.vp_main);
        tabLayout = findViewById(R.id.tl_main);

        fragments.add(MojiFragment.newInstance("happy"));
        fragments.add(MojiFragment.newInstance("sad"));
        fragments.add(MojiFragment.newInstance("angry"));
        fragments.add(MojiFragment.newInstance("doubt"));
        titles.add("happy");
        titles.add("sad");
        titles.add("angry");
        titles.add("doubt");

        adapter = new MyPagerAdapter(getSupportFragmentManager());
        adapter.setFragments(fragments);
        adapter.setTitles(titles);
        viewPager.setAdapter(adapter);
        tabLayout.setupWithViewPager(viewPager);

    }
}

File: jie_yan/homework_summer/EyeP/app/src/main/java/pow/jie/eyep/MainActivity.kt
package pow.jie.eyep

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.activity_main.*
import pow.jie.eyep.adapter.MainFragmentAdapter


class MainActivity : AppCompatActivity() {

    private val onNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.navigation_home -> {
                view_pager_main.currentItem = 0
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_rank -> {
                view_pager_main.currentItem = 1
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_search -> {
                view_pager_main.currentItem = 2
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    private val onPageChangeListener = object :OnPageChangeListener {
        override fun onPageScrollStateChanged(state: Int) {}

        override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}

        override fun onPageSelected(position: Int) {
            nav_view.menu.getItem(position).isChecked = true
        }
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        nav_view.setOnNavigationItemSelectedListener(onNavigationItemSelectedListener)
        view_pager_main.addOnPageChangeListener(onPageChangeListener)
        val adapter = MainFragmentAdapter(supportFragmentManager)
        view_pager_main.adapter = adapter

    }

}

